client/lib/posts/post-list-cache-store.js

```diff
@@ -1,71 +1,66 @@
 /**
  * External dependencies
  */
-var intersection = require( 'lodash/intersection' );
+import intersection from 'lodash/intersection';
+import debugFactory from 'debug';
 
 /**
  * Internal dependencies
  */
-var sites = require( 'lib/sites-list' )(),
-	Dispatcher = require( 'dispatcher' );
-
-var _cache = {},
-	TTL_IN_MS = 5 * 60 * 1000; // five minutes
+import sitesFactory from 'lib/sites-list';
+import Dispatcher from 'dispatcher';
+
+let _cache = {};
+const _canonicalCache = {};
+const TTL_IN_MS = 5 * 60 * 1000; // five minutes
+const sites = sitesFactory();
+const PostsListCache = {
+	get,
+	_reset: function() {
+		_cache = {};
+	}
+};
+const debug = debugFactory( 'calypso:posts-list:cache' );
 
 function isStale( list ) {
-	var now = new Date().getTime(),
-		timeSaved = list.timeSaved;
+	const now = new Date().getTime();
+	const { timeSaved } = list;
 	return ( now - timeSaved ) > TTL_IN_MS;
 }
 
-function getCacheKey( options ) {
-	var cacheKey = '',
-		keys = Object.keys( options ).sort();
-
-	keys.forEach( function( key ) {
-		if ( cacheKey.length ) {
-			cacheKey += ':';
-		}
-
-		cacheKey += key + '-' + options[ key ];
-	} );
-
-	return cacheKey;
-}
-
-function get( query ) {
-	var key = getCacheKey( query );
-
-	if ( _cache[ key ] && ! isStale( _cache[ key ] ) && ! _cache[ key ].dirty ) {
-		return _cache[ key ].list;
+function get( listKey ) {
+	if ( _cache[ listKey ] && ! isStale( _cache[ listKey ] ) && ! _cache[ listKey ].dirty ) {
```

Comment:
This seems long enough to move into a helper check, maybe something like
`isListKeyFresh`